tracer/test/Datadog.Trace.TestHelpers/NativeProcess/CreateProcess.cs

```diff
@@ -0,0 +1,358 @@
+// <copyright file="CreateProcess.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+// Based on https://github.com/dotnet/runtime/blob/045e55abac82e89e0daaee47bcb8433e0fc9ccbc/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+namespace Datadog.Trace.TestHelpers.NativeProcess;
+
+public class CreateProcess
+{
+    private static readonly object CreateProcessLock;
+
+    static CreateProcess()
+    {
+        // Just making extra sure that the Process static constructor has run
+        _ = Process.GetCurrentProcess();
+
+#if NETFRAMEWORK
+        const string FieldName = "s_CreateProcessLock";
+#else
+        const string FieldName = "s_createProcessLock";
+#endif
+
+        var createProcessLock = typeof(Process)
+            .GetField(FieldName, System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic)!
+            .GetValue(null);
+
+        CreateProcessLock = createProcessLock ?? throw new InvalidOperationException($"Failed to read the {FieldName} field from Process");
+    }
+
+    internal static SuspendedProcess StartSuspendedProcess(ProcessStartInfo startInfo)
+    {
+        // See knowledge base article Q190351 for an explanation of the following code.  Noteworthy tricky points:
```

Comment:
Is there a link to the article?

Reply:
Oh. No, that's copy/pasted from Microsoft's code